Telemetry validation breaks build on main: "Property sessionid on event vscodeAgents.sessions/requestSent is overloaded by a common property"

The `Generate lists of telemetry events` step in the Linux product build is failing on `main`, which blocks PR CI for everyone (including unrelated PRs that just merge the latest `main`).

### What's failing

From the failing pipeline step `Generate lists of telemetry events` (`node build/azure-pipelines/common/extract-telemetry.ts`):

```
Property sessionid on event vscodeAgents.sessions/requestSent is overloaded by a common property
Error: Validation failed, please see logs for more information
    at extractAndResolveDeclarations (.../@vscode/telemetry-extractor/out/lib/save-declarations.js:166:19)
Telemetry extraction failed: Command failed: node ".../@vscode/telemetry-extractor/out/extractor.js" ...
##[error]Bash exited with code '1'.
```

### Root cause

Commit 8bfc82f5809 ("sessions improvements", merged via #318435) added a new telemetry event `vscodeAgents.sessions/requestSent` in `src/vs/sessions/services/sessions/browser/sessionsTelemetryReporter.ts`. Its classification declares a property named `sessionId`:

https://github.com/microsoft/vscode/blob/main/src/vs/sessions/services/sessions/browser/sessionsTelemetryReporter.ts#L27-L34

and the `publicLog2` call passes that same `sessionId` in the payload:

https://github.com/microsoft/vscode/blob/main/src/vs/sessions/services/sessions/browser/sessionsTelemetryReporter.ts#L218

`@vscode/telemetry-extractor` compares event property names against the common (GDPR) properties case-insensitively, and `sessionID` is already a reserved common property attached to every event:

- `__GDPR__COMMON__ "sessionID"` in `src/vs/platform/telemetry/common/commonProperties.ts`
- Wired in via `TelemetryService` (`this.sessionId = this._commonProperties['sessionID']`)

So the new event's `sessionId` lowercases to `sessionid` and collides with the common `sessionID`, which the extractor rejects with the `overloaded by a common property` error — hence the build failure.

### Suggested fix (Copilot generated) 

Rename the event-local property to something that doesn't collide with the reserved `sessionID` (e.g. `agentSessionId`) in both the classification type and the two `publicLog2` payloads in `sessionsTelemetryReporter.ts` (around lines 27/34, 218, and 260).

### Impact

`Generate lists of telemetry events` is required for the Linux product build, so this blocks PR validation for unrelated changes until it's fixed on `main`.

cc @benibenj

--- a/src/vs/sessions/services/sessions/browser/sessionsTelemetryReporter.ts
+++ b/src/vs/sessions/services/sessions/browser/sessionsTelemetryReporter.ts
@@ -24,14 +24,14 @@ type SessionIsolationKind = 'worktree' | 'folder';
 // --- Field group: session (derived from ISession) ---
 
 type SessionFields = {
-	sessionId: string;
+	agentSessionId: string;
 	providerId: string;
 	providerType: string;
 	chatCount: number;
 };
 
 type SessionFieldsClassification = {
-	sessionId: { classification: 'SystemMetaData'; purpose: 'FeatureInsight'; comment: 'Globally unique session id (providerId:resourceUri), used to correlate events for the same session.' };
+	agentSessionId: { classification: 'SystemMetaData'; purpose: 'FeatureInsight'; comment: 'Globally unique session id (providerId:resourceUri), used to correlate events for the same session.' };
 	providerId: { classification: 'SystemMetaData'; purpose: 'FeatureInsight'; comment: 'The sessions provider identifier (e.g., remote agent host or local).' };
 	providerType: { classification: 'SystemMetaData'; purpose: 'FeatureInsight'; comment: 'The session type identifier provided by the sessions provider.' };
 	chatCount: { classification: 'SystemMetaData'; purpose: 'FeatureInsight'; isMeasurement: true; comment: 'Number of chats currently in the session.' };
@@ -257,7 +257,7 @@ export class SessionsTelemetryReporter extends Disposable {
 
 	private _getSessionFields(session: ISession): SessionFields {
 		return {
-			sessionId: session.sessionId,
+			agentSessionId: session.sessionId,
 			providerId: session.providerId,
 			providerType: session.sessionType,
 			chatCount: session.chats.get().length,

